build-tools/packages/build-tools/src/type-test-generator/compatibility.ts

```diff
@@ -25,10 +26,13 @@ type requireAssignableTo<_A extends B, B> = true;
  * This will strip some type branding information which ideally would be kept for stricter checking, but without it, const enums show up as breaking when unchanged.
  */
 export const typeOnly = `
+// See 'build-tools/src/type-test-generator/compatibility.ts' for more information.
 type TypeOnly<T> = T extends number
 	? number
 	: T extends string
 	? string
+	: T extends boolean | bigint | symbol
```

Comment:
Including number string here would actually make a difference. If we have a type like `number | string` right now it fill fall through to the object case below, but we could fix that and allow it to keep that type information by including it here.

Reply:
This didn't work the way I expected.

'TypeOnly<string | number>' appears to be decomposed and processed as 'TypeOnly<string> | TypeOnly<number>', which hits the earlier cases that strip nominal typing.

(I added a test cases below to illustrate / verify.)